### DISCUSSION TITLE
Adding additional (custom ?) button/component to TopNavigation component

### DISCUSSION THREAD

**Discussion by mguja-fcc** [cloudscape-design/components#581] [Q&A] [answered]
I need help adding my own dropdown button that a service can be chosen in the top navigation component. Currently I only see identity, search and utilities (stacks everything on the right). I want to "mimic" AWS console and have services button (could be dropdown) on the left, right after identity. MYBUTTON is where I want to add my button to choose a service. How can this be accomplished ? Thanks.

```
<TopNavigation
          i18nStrings={i18nStrings}
          identity={{
            href: '#',
            title: 'My Test App',
             logo: { src: logo, alt: 'Service name logo' },
          }}

         MYBUTTON={

                }
           search={
             <Input
               ariaLabel="Input field"
               value={searchValue}
               type="search"
               placeholder="Search"
               onChange={({ detail }) => setSearchValue(detail.value)}
             />
           }

          utilities={[
            {
               type: 'button',
               iconName: 'notification',
               ariaLabel: 'Notifications',
               badge: true,
               disableUtilityCollapse: true,
             },
             { type: 'button', iconName: 'settings', title: 'Settings', ariaLabel: 'Settings' },
            {
              type: 'menu-dropdown',
              text: 'Customer name',
              description: 'customer@example.com',
              iconName: 'user-profile',
              items: profileActions,
            },
          ]}
        />
```

**Comment by just-boris** [comment, marked as answer]
Hello!

Our top navigation component does not have the intent to replicate AWS Console navigation bar. Some features may not be in par between these two.

If you want to use our top navigation component, dropdown menus can be placed only to the right there

**Comment by t0mas** [comment]
@mguja-fcc did you take another approach to this?

**Reply by hopkinsnji** [reply]
I have the same issue, I prefer the aws console top navigation because the height is minimal which makes it nice and compact. The top nav in CloudScape it a bit too thick, I'll open a feature request and see if it can be added as a different component or changed.
